contract/test/test-contract.js

```diff
@@ -82,38 +123,63 @@ test('Start the contract', async t => {
  * @param {Purse} purse
  * @param {string[]} choices
  */
-const alice = async (t, zoe, instance, purse, choices = ['map', 'scroll']) => {
+const alice = async (
+  t,
+  zoe,
+  instance,
+  purse,
+  choices = ['frontRow', 'middleRow'],
+) => {
   const publicFacet = E(zoe).getPublicFacet(instance);
   // @ts-expect-error Promise<Instance> seems to work
   const terms = await E(zoe).getTerms(instance);
-  const { issuers, brands, tradePrice } = terms;
+  const { issuers, brands } = terms;
 
   const choiceBag = makeCopyBag(choices.map(name => [name, 1n]));
+  const totalPrice = bagPrice(choiceBag, terms.inventory);
   const proposal = {
-    give: { Price: tradePrice },
-    want: { Items: AmountMath.make(brands.Item, choiceBag) },
+    give: { Price: totalPrice },
+    want: { Tickets: AmountMath.make(brands.Ticket, choiceBag) },
   };
-  const pmt = await E(purse).withdraw(tradePrice);
+  const pmt = await E(purse).withdraw(totalPrice);
   t.log('Alice gives', proposal.give);
-  // #endregion makeProposal
 
   const toTrade = E(publicFacet).makeTradeInvitation();
 
   const seat = E(zoe).offer(toTrade, proposal, { Price: pmt });
-  const items = await E(seat).getPayout('Items');
+  const tickets = await E(seat).getPayout('Tickets');
 
-  const actual = await E(issuers.Item).getAmountOf(items);
+  const actual = await E(issuers.Ticket).getAmountOf(tickets);
   t.log('Alice payout brand', actual.brand);
   t.log('Alice payout value', actual.value);
-  t.deepEqual(actual, proposal.want.Items);
+  t.deepEqual(actual, proposal.want.Tickets);
+};
+
+const makeTerms = brand => {
+  return {
+    inventory: {
+      frontRow: {
+        tradePrice: AmountMath.make(brand, 3n),
```

Comment:
Note that IST has 6 decimal places. So 1 IST is `AmountMath.make(istBrand, 1_000_000n)`. so this function might want to take a unit amount instead of just a brand.